x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankRetrieverBuilder.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.rank.textsimilarity;
+
+import org.elasticsearch.license.LicenseUtils;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.retriever.RetrieverBuilder;
+import org.elasticsearch.search.retriever.RetrieverParserContext;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xpack.core.XPackPlugin;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.search.rank.RankBuilder.DEFAULT_RANK_WINDOW_SIZE;
+
+/**
+ * A {@code RetrieverBuilder} for parsing and constructing a text similarity reranker retriever.
+ */
+public class TextSimilarityRankRetrieverBuilder extends RetrieverBuilder {
+
+    public static final ParseField RETRIEVER_FIELD = new ParseField("retriever");
+    public static final ParseField FIELD_FIELD = new ParseField("field");
+    public static final ParseField RANK_WINDOW_SIZE_FIELD = new ParseField("rank_window_size");
+    public static final ParseField INFERENCE_ID_FIELD = new ParseField("inference_id");
+    public static final ParseField INFERENCE_TEXT_FIELD = new ParseField("inference_text");
+    public static final ParseField MIN_SCORE_FIELD = new ParseField("min_score");
+
+    public static final ObjectParser<TextSimilarityRankRetrieverBuilder, RetrieverParserContext> PARSER = new ObjectParser<>(
+        TextSimilarityRankBuilder.NAME,
+        TextSimilarityRankRetrieverBuilder::new
+    );
+
+    static {
+        PARSER.declareObject((r, v) -> r.retrieverBuilder = v, (p, c) -> {
+            RetrieverBuilder retrieverBuilder = RetrieverBuilder.parseInnerRetrieverBuilder(p, c);
+            c.trackSectionUsage(TextSimilarityRankBuilder.NAME + ":" + RETRIEVER_FIELD.getPreferredName());
```

Comment:
We shouldn't do this track usage for all these fields individually. Its gonna add so much noise. 

Right now we only care about the similarity ranker being used, we don't care if folks set window size, etc. This is really overkill.

Reply:
Removed all tracking from here. The retriever builder already tracks the usage on specific retrievers, e.g. `standard` or `text_similarity_reranker`.